### ISSUE TITLE
Playlist should confirm deletion of playlists, per Figma

### ISSUE THREAD

**Issue by stephendonner** [brave/brave-browser#40248]
### Description

Playlist should confirm deletion of playlists, per Figma

### Steps to reproduce

1. install `1.70.41`
2. launch Brave
3. restart Brave (to pick up Griffin seeds for Playlist)
4. add a media file/video to your playlist
5. select the media file
6. hover over and click on the `...` 3-dots menu
7. choose `Remove from playlist`

### Actual result

Nothing visually happens (other than the media file is instantly deleted)

### Expected result

example | example
---------|---------
<img width="1106" alt="Screen Shot 2024-08-02 at 5 57 17 PM" src="https://github.com/user-attachments/assets/54b8577b-79fc-4e91-9ecf-15b3ca0733aa"> | <img width="381" alt="Screen Shot 2024-08-02 at 5 57 25 PM" src="https://github.com/user-attachments/assets/823c7379-5c5c-4aa5-81ad-acefa22c8c36">

### Reproduces how often

Easily reproduced

### Brave version (brave://version info)

```
<img width="381" alt="Screen Shot 2024-08-02 at 5 57 25 PM" src="https://github.com/user-attachments/assets/d07948fd-12c0-4ffd-8748-ab1d7f8c3b4b">
Brave | 1.70.41 Chromium: 127.0.6533.88 (Official Build) nightly (x86_64)
-- | --
Revision | 9303bae9fe07bc2ec7c0cdef139e87304b2d6c96
OS | macOS Version 11.7.10 (Build 20G1427)
```

### Channel information

- [ ] release (stable)
- [X] beta
- [X] nightly

### Reproducibility

- [X] with Brave Shields disabled
- [X] with Brave Rewards disabled
- [ ] in the latest version of Chrome

### Miscellaneous information

cc @aguscruiz @rebron @sangwoo108 @szilardszaloki @brave/qa-team

**Comment by sangwoo108** [comment]
Hi, could you try removing Playlist folder, not individual media items? I thought the warning message is for that. cc @aguscruiz

**Comment by aguscruiz** [comment]
I wouldn't be against adding a dialog for deleting individual items too, although may be a bit annoying after a while. What do you think @rebron ?

**Comment by stephendonner** [comment]
> Hi, could you try removing Playlist folder, not individual media items? I thought the warning message is for that. cc @aguscruiz

Thanks, you're correct.  However, that doesn't prompt either.

**Comment by stephendonner** [comment]
Not sure if this prompt should look like Figma?

<img width="1694" alt="Screen Shot 2024-08-05 at 10 41 33 AM" src="https://github.com/user-attachments/assets/7935e221-3b93-4de7-83f2-f20d6188cad8">

**Comment by aguscruiz** [comment]
Yeah, there should be a maximum width established. I think I used the same width as other dialogs that appear in the browser. But this one is too wide. 

I have a 512px max limit set up on the file

**Comment by rebron** [comment]
> I wouldn't be against adding a dialog for deleting individual items too, although may be a bit annoying after a while. What do you think @rebron ?

If we can hook up Undo, I don't think we need a delete confirmation dialog for individual items.
